From d595779341387bf7c867ac28811883a3b8fe6e88 Mon Sep 17 00:00:00 2001
From: Julien Cristau <jcristau@debian.org>
Date: Sun, 29 Apr 2012 16:27:52 +0200
Subject: [PATCH libX11] configure: check if issetugid is declared

GNU/kFreeBSD has issetugid in libc (for legacy apps?), but doesn't
declare it anywhere, causing gcc to error out with
-Werror=implicit-function-declaration.  Use AC_CHECK_DECL in addition to
AC_CHECK_FUNC so we disable this code instead of failing to build it.

Debian bug#669670 <http://bugs.debian.org/669670>

Signed-off-by: Julien Cristau <jcristau@debian.org>
---
 configure.ac |    5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/configure.ac b/configure.ac
index aaec968..c91234f 100644
--- a/configure.ac
+++ b/configure.ac
@@ -223,8 +223,9 @@ AC_CHECK_HEADERS([sys/select.h])
 AC_CHECK_FUNCS([strtol seteuid])
 # Used in lcFile.c (see also --enable-xlocaledir settings below)
 XLOCALEDIR_IS_SAFE="no"
-AC_CHECK_FUNC([issetugid], [XLOCALEDIR_IS_SAFE="yes"]
-	AC_DEFINE(HASSETUGID,1,[Has issetugid() function]))
+AC_CHECK_DECL([issetugid],
+	AC_CHECK_FUNC([issetugid], [XLOCALEDIR_IS_SAFE="yes"]
+		AC_DEFINE(HASSETUGID,1,[Has issetugid() function])))
 AC_CHECK_FUNC([getresuid], [XLOCALEDIR_IS_SAFE="yes"]
 	AC_DEFINE(HASGETRESUID,1,[Has getresuid() & getresgid() functions]))
 # Used in Font.c
-- 
1.7.10

